odoo/models.py

```diff
@@ -6709,6 +6721,39 @@ def flush_recordset(self, fnames=None):
         if self.env.cache.has_dirty_fields(self, fields_):
             self._flush(fnames)
 
+    def _ensure_exists(self):
+        if not self:
+            return
+        precommit = self.env.cr.precommit
+        if 'ensure_exists' not in precommit.data:
+            precommit.data['ensure_exists'] = defaultdict(set)
+            precommit.add(self._ensure_exists_callback)
+        precommit.data['ensure_exists'][self._name].update(self._ids)
+
+    def _ensure_exists_callback(self):
+        """ Enusre records' existence in the current and all concurrent transactions """
+        data = self.env.cr.precommit.data.get('ensure_exists')
+        if not data:
+            return
+        for model_name, ids in data.items():
+            if not ids:
+                continue
+            model = self.env[model_name]
+            # Lock referenced records by performing a dummy update to prevent concurrent deletion
+            self.env.cr.execute(SQL("""
+                UPDATE %(table)s
+                SET id = id
+                WHERE id IN %(ids)s
```

Comment:
@rco-odoo what about?
```suggestion
                WHERE id IN %(ids)s
                AND xmin != txid_current()::text::xid,
```